server/flaskr/blueprints/role.py

```
@@ -27,36 +27,42 @@ def get_role(id: int):
+def update_role(roleId: str, permission_name, permission_value):
+    if permission_name == "manageTeam" and not permission_value:
+        if UserDataHandler.get_user_role_id(g.user_id) == roleId:
+            return jsonify({"error": "self_lockout"}), 403
+
+    RoleDataHandler.update_role(roleId, permission_name, permission_value)
     return "", 204
 
 @roles_bp.delete("/<int:id>")
-@permission_middleware(Permission.CanUpdatePermissions)
+@permission_middleware(Permission.ManageTeam)
 def delete_role(id: int):
```

Ruff (0.15.20)

[error] Function argument `id` is shadowing a Python builtin

(A002)